### DISCUSSION TITLE
Mesh refinement of two-dimensional polycrystalline model

### DISCUSSION THREAD

**Discussion by ZJM-19491001** [neperfepx/neper#810] [Q&A] [answered]
Dear friends:
I 'm a newcomer to neper. I want to know how to divide a region or a grain into denser grids in a two-dimensional polycrystalline model, while the other parts use coarse grids. I have tried -domface/cut orders... But it was not successful, besides, I need CPE4 element.
I 'd appreciate it if someone could answer that question! Thank you very much!

**Comment by rquey** [comment, marked as answer]
You can define the element size using a custom `-rcl` value. For example, to refine grain 1:
```
neper -T -n 100 -dim 2 -reg 1
neper -M n100-id1.tess -rcl "((id<=1)?0.2:2)" -pl 10
neper -V n100-id1.tess,n100-id1.msh -dataelt2dedgerad 0.001 -dataelt1drad 0.003 -showelt1d all -print d
```
![d](https://github.com/neperfepx/neper/assets/2659376/7cfc48c3-920a-4c47-b73f-94930c44d607)

Neper only generates triangular elements in 2D, but you should be able to get quads as described in [this discussion](https://github.com/neperfepx/neper/discussions/98).

**Reply by ZJM-19491001** [reply]
Dear Dr. quey :

Thank you for your reply and thank you for your hard work for neper!

Yours, sincerely.
